tripal/src/Services/TripalPublish.php

```diff
@@ -476,12 +476,20 @@ protected function getEntityTitles($matches) {
           $delta = 0;
           $field = $this->field_info[$field_name]['instance'];
           $main_prop = $field->mainPropertyName();
-          $value = $match[$field_name][$delta][$main_prop]['value']->getValue();
-          if ($value !== NULL) {
-            $entity_title = trim(preg_replace("/\[$field_name\]/", $value,  $entity_title));
+          $value = '';
+          if (array_key_exists($delta, $match[$field_name])) {
+            $value = $match[$field_name][$delta][$main_prop]['value']->getValue();
           }
+          if ($value === NULL) {
+            $value = '';
+          }
+          $entity_title = trim(preg_replace("/\[$field_name\]/", $value,  $entity_title));
         }
       }
+      // Trim any training spaces and remove double spaces. Double spaces
+      // can occur if a token replacment has no value but there are spaces
+      // around it.
+      $entity_title = trim(preg_replace('/\s\s/', ' ', $entity_title));
```

Comment:
```suggestion
      // Trim any training spaces and remove double spaces. Double spaces
      // can occur if a token replacment has no value but there are spaces
      // around it.
      $entity_title = trim(preg_replace('/\s\s/', ' ', $entity_title));
```
```suggestion
      // Trim any trailing spaces and remove double spaces. Double spaces
      // can occur if a token replacement has no value but there are spaces
      // around it.
      $entity_title = trim(preg_replace('/\s\s+/', ' ', $entity_title));
```
I am suggesting this regex change in the unlikely chance there are more than two spaces, it will still collapse to one.